gcode.c

```diff
@@ -80,368 +78,611 @@ static uint8_t gc_check_same_position(float *pos_a, float *pos_b)
+    case 92: 
+      switch(mantissa) {
+        case 0: 
+          gc_block.non_modal_command = NON_MODAL_SET_COORDINATE_OFFSET;
+          break; // G92
+        case 10: 
+          gc_block.non_modal_command = NON_MODAL_RESET_COORDINATE_OFFSET;
+          break; // G92.1
+        default:
+          // [Unsupported G92.x command]
+          FAIL(STATUS_GCODE_UNSUPPORTED_COMMAND);      }
+      mantissa = 0; // Set to zero to indicate valid non-integer G command.
+      break;      
+  }
+  return STATUS_GCODE_NO_FAIL; 
+}
+
+uint8_t gc_process_modal_group_G1(uint8_t int_value, uint8_t mantissa, uint8_t * axis_command)
+{
+  switch(int_value) {
+    case 0: case 1: case 2: case 3: case 38: 
+    // Check for G0/1/2/3/38 being called with G10/28/30/92 on same block.
+    // * G43.1 is also an axis command but is not explicitly defined this way.
+      if (*axis_command) { FAIL(STATUS_GCODE_AXIS_COMMAND_CONFLICT); } // [Axis word/command conflict]
+      *axis_command = AXIS_COMMAND_MOTION_MODE; 
+      // No break. Continues to next line.
+    case 80: 
+      //word_bit = MODAL_GROUP_G1; 
+      switch(int_value) {
+        case 0: gc_block.modal.motion = MOTION_MODE_SEEK; break; // G0
+        case 1: gc_block.modal.motion = MOTION_MODE_LINEAR; break; // G1
+        case 2: gc_block.modal.motion = MOTION_MODE_CW_ARC; break; // G2
+        case 3: gc_block.modal.motion = MOTION_MODE_CCW_ARC; break; // G3
+        case 38: 
+          switch(mantissa) {
+            case 20:  // G38.2
+              gc_block.modal.motion = MOTION_MODE_PROBE; 
+              break;
+            // NOTE: If G38.3+ are enabled, change mantissa variable type to uint16_t.
+            // case 30: gc_block.modal.motion = MOTION_MODE_PROBE_NO_ERROR; break; // G38.3 Not supported.
+            // case 40: // Not supported.
+            // case 50: // Not supported.
+            default: FAIL(STATUS_GCODE_UNSUPPORTED_COMMAND); // [Unsupported G38.x command]
+          }
+          mantissa = 0; // Set to zero to indicate valid non-integer G command.
+          break;
+        case 80: gc_block.modal.motion = MOTION_MODE_NONE; break; // G80
+      }            
+      break;
+  }
+  return STATUS_GCODE_NO_FAIL;
+}
+
+uint8_t gc_process_modal_group_G2(uint8_t int_value, uint8_t mantissa, uint8_t * axis_command)
+{
+  (void)mantissa;
+  (void)axis_command;
+
+  switch(int_value) {
+    case 17:
+      gc_block.modal.plane_select = PLANE_SELECT_XY;
+      break;
+    case 18:
+      gc_block.modal.plane_select = PLANE_SELECT_ZX;
+      break;
+    case 19:
+      gc_block.modal.plane_select = PLANE_SELECT_YZ;
+      break;
+  }
+  return STATUS_GCODE_NO_FAIL;
+}
+
+uint8_t gc_process_modal_group_G3(uint8_t int_value, uint8_t mantissa, uint8_t * axis_command)
+{
+  (void)mantissa;
+  (void)axis_command;
+
+  if (int_value == 90)
+    gc_block.modal.distance = DISTANCE_MODE_ABSOLUTE;  // G90
+  else
+    gc_block.modal.distance = DISTANCE_MODE_INCREMENTAL;  // G91
+
```

Comment:
I try to stick to the kernel style. I agree with the reasoning in the kernel doc.
http://lxr.free-electrons.com/source/Documentation/CodingStyle Lines 169-202

I think it especially holds true in a file such as this one, where a lot of newlines are used already because of the case statements. 

P.S. The original grbl writers tried to minimize the newlines by doing horrible things like `case: statement; break;`. I missed some of these statements - thanks for pointing them out.